### PR TITLE
feat(addon-editor): add `@tiptap/extension-youtube`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Part of #3244

## What is the new behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/addon-editor/directives/tiptap-editor/tiptap-editor.service.ts [hunk 1]
```diff
@@ -343,4 +344,8 @@ export class TuiTiptapEditorService extends AbstractTuiEditor {
     setFileLink(preview: TuiEditorAttachedFile): void {
         this.editor.commands.setFileLink(preview);
     }
+
+    setYoutubeVideo(options: TuiYoutubeOptions): void {
+        this.editor.commands.setYoutubeVideo(options as any);
```
Comment (waterplea) [inline projects/addon-editor/directives/tiptap-editor/tiptap-editor.service.ts:349]: Why different interface?

File: projects/demo/src/modules/app/pages.ts [hunk 2]
```diff
@@ -936,6 +936,12 @@ export const pages: TuiDocPages = [
         keywords: `editor, markdown, wysiwyg, редактор, текст, html, rich, text`,
         route: `/editor/markdown`,
     },
+    {
+        section: $localize`Editor`,
+        title: `Embed`,
+        keywords: `editor, embed, video player, audio player, wysiwyg, редактор, плейер, плеер, текст, html, rich, text`,
```
Comment (waterplea) [inline projects/demo/src/modules/app/pages.ts:942]: youtube

**Review by waterplea — APPROVED** [review]

File: projects/addon-editor/directives/tiptap-editor/tiptap-editor.service.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-editor/directives/tiptap-editor/tiptap-editor.service.ts:349]: I want to pass not only number of pixels, but percentages `width="100%"`. if you pass that value, it works, but for some reason their interface only supports numbers

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/demo/src/modules/app/pages.ts (hunk 2 above)
Comment (splincode) [reply projects/demo/src/modules/app/pages.ts]: fixed

**Pull request merged**